File: app/src/main/java/com/stevenfrew/beatprompter/cloud/CloudItemInfo.java
package com.stevenfrew.beatprompter.cloud;

public abstract class CloudItemInfo {
    // Unique ID of the item in the cloud storage.
    public String mID;
    // Display name.
    public String mName;

    CloudItemInfo(String id,String name)
    {
        mID=id;
        mName=name;
    }

    int compareTo(CloudItemInfo other)
    {
        boolean thisIsFolder=this instanceof CloudFolderInfo;
        boolean otherIsFolder=other instanceof CloudFolderInfo;
        if(thisIsFolder && !otherIsFolder)
            return -1;
        if(!thisIsFolder && otherIsFolder)
            return 1;
        return mName.toLowerCase().compareTo(other.mName.toLowerCase());
    }
}

File: app/src/main/java/com/stevenfrew/beatprompter/cloud/CloudItemInfo.kt
package com.stevenfrew.beatprompter.cloud

abstract class CloudItemInfo internal constructor(// Unique ID of the item in the cloud storage.
        var mID: String, // Display name.
        var mName: String):Comparable<CloudItemInfo> {

    override operator fun compareTo(other: CloudItemInfo): Int {
        val thisIsFolder = this is CloudFolderInfo
        val otherIsFolder = other is CloudFolderInfo
        if (thisIsFolder && !otherIsFolder)
            return -1
        return if (!thisIsFolder && otherIsFolder) 1 else mName.toLowerCase().compareTo(other.mName.toLowerCase())
    }
}